File: empowery_co_op/co_op/report/supplier_engagement/supplier_engagement.js
// Copyright (c) 2016, GreyCube Technologies and contributors
// For license information, please see license.txt
/* eslint-disable */
var color_codes = {}
frappe.query_reports["Supplier Engagement"] = {

  "filters": [
  ],
  onload: (report) => {
    frappe.call({
      method: "empowery_co_op.co_op.report.supplier_engagement.supplier_engagement.get_color_codes",
      args: {},
      callback: function (data) {
        color_codes = data.message
      }
    })

  },
  "formatter": function(value, row, column, data, default_formatter) {
  // "formatter": function (row, cell, value, columnDef, dataContext,column,data, default_formatter) {
    value = default_formatter(value, row, column, data);

    if (column.name == '% of Engagement') {
      if (1==1) {
        
        per_of_engagement = null
        href_start_index = value.search('\">');
        href_end_index = value.search('</a>');
        per_of_engagement = parseFloat(value.slice(href_start_index + 2, href_end_index))
        console.log(per_of_engagement)
        if (per_of_engagement != null || per_of_engagement != '') {

          if ((per_of_engagement >= 0.0) && (per_of_engagement <= parseFloat(color_codes.yellow_start))) {
            value = `<p style='margin:0px;padding-left:5px;background-color:${color_codes.color_1}!important;'>${value}</p>`
          } else if (per_of_engagement > parseFloat(color_codes.yellow_start) && per_of_engagement <= parseFloat(color_codes.yellow_end)) {
            value = `<p style='margin:0px;padding-left:5px;background-color:${color_codes.color_2}!important;'>${value}</p>`
          } else if (per_of_engagement > parseFloat(color_codes.yellow_end) && per_of_engagement <= 100.00) {
            value = `<p style='margin:0px;padding-left:5px;background-color:${color_codes.color_3}!important;'>${value}</p>`
          }
          return value;
        }
        return value;
      }
    }
    return value;
  }
}

File: empowery_co_op/co_op/report/supplier_engagement/supplier_engagement.py
# Copyright (c) 2013, GreyCube Technologies and contributors
# For license information, please see license.txt

from __future__ import unicode_literals
import frappe
from frappe import _

def execute(filters=None):
	columns, data = get_columns(), get_data()
	return columns, data

def get_columns():
		return [
		{
			"fieldname": "Supplier",
			"label": _("Supplier"),
			"fieldtype": "Link",
			"options":"Supplier",
			"width": 300
		},
		{
			"fieldname": "# of Linked Customer",
			"label": _("# of Linked Customer"),
			"fieldtype": "Data",
			"width": 150
		},
		{
			"fieldname": "% of Engagement",
			"label": _("% of Engagement"),
			"fieldtype": "Data",
			"width": 150
		}]
	

def get_data():
	data = frappe.db.sql("""
		select concat('<a href="desk#query-report/Customer%20Engagement?supplier=',supp.name,'">',supp.name,'</a>') as 'Supplier',
				concat('<a href="desk#query-report/Customer%20Engagement?supplier=',supp.name,'">',cust_link.linked,'</a>')as '# of Linked Customer',
				concat('<a href="desk#query-report/Customer%20Engagement?supplier=',supp.name,'">',ROUND(cust_link.linked/(select count(customer_name) from `tabCustomer` where docstatus=0 and customer_group != 'Supplier')*100,2),'</a>')as '% of Engagement'
    	from tabSupplier supp
    		left outer join (
    			select parent,count(*) linked from `tabDynamic Link` 
    			where link_doctype='Customer' and parenttype='Supplier'and docstatus = 0 and parentfield='links' 
    			group by parent) cust_link on cust_link.parent = supp.name
    			where supp.docstatus=0 and supp.disabled!=1""", as_dict=1)
		
	return data
@frappe.whitelist()
def get_color_codes():
	color_codes={}
	color_codes['color_1']=frappe.db.get_single_value('Supplier Engagement Report Settings', 'color_1')
	color_codes['red_start']=frappe.db.get_single_value('Supplier Engagement Report Settings', 'red_zone_start_value')
	color_codes['color_2']=frappe.db.get_single_value('Supplier Engagement Report Settings', 'color_2')
	color_codes['yellow_start']=frappe.db.get_single_value('Supplier Engagement Report Settings', 'yellow_zone_start_value')
	color_codes['yellow_end']=frappe.db.get_single_value('Supplier Engagement Report Settings', 'yellow_zone_end_value')
	color_codes['color_3']=frappe.db.get_single_value('Supplier Engagement Report Settings', 'color_3')
	color_codes['green_end']=frappe.db.get_single_value('Supplier Engagement Report Settings', 'green_zone_end_value')
	return color_codes